packages/fleather/lib/src/rendering/embed_proxy.dart

```diff
@@ -50,3 +50,114 @@ class RenderEmbedProxy extends RenderProxyBox implements RenderContentProxyBox {
   @override
   double get preferredLineHeight => size.height;
 }
+
+/// Proxy to an arbitrary embeddable [RenderBox].
+///
+/// Computes necessary editing metrics based on the dimensions of the childlren
+/// render boxes.
+class RenderGroupEmbedProxy extends RenderFlex
+    implements RenderContentProxyBox {
+  RenderGroupEmbedProxy({
+    TextDirection? textDirection,
+  }) : super(textDirection: textDirection);
+
+  @override
+  List<TextBox> getBoxesForSelection(TextSelection selection) {
+    if (selection.isCollapsed) {
+      return [];
+    }
+
+    final childIter = _childrenIterator();
+    final accWidth = childIter
+        .take(selection.extentOffset)
+        .fold(0.0, (acc, child) => acc + child.size.width);
+
+    return <TextBox>[
+      TextBox.fromLTRBD(0.0, 0.0, accWidth, size.height, TextDirection.ltr)
+    ];
+  }
+
+  @override
+  double? getFullHeightForCaret(TextPosition position) {
+    return preferredLineHeight;
+  }
+
+  @override
+  Offset getOffsetForCaret(TextPosition position, Rect caretPrototype) {
+    final childIter = _childrenIterator();
+    final accWidth = childIter
+        .take(position.offset)
+        .fold(0.0, (acc, child) => acc + child.size.width);
+    return Offset(accWidth, 0.0);
+  }
+
+  @override
+  TextPosition getPositionForOffset(Offset offset) {
+    if (childCount == 0) {
+      return const TextPosition(offset: 0);
+    }
+
+    double accWidth = 0;
+    int i = 0;
+    final childIter = _childrenIterator();
+
+    for (final child in childIter) {
```

Comment:
Probably make sense to add the length of the child instead of assuming it’s 1. 
i wonder if all those functions are not already written in block or editor.